code/modules/vehicles/speedbike.dm

```diff
@@ -24,6 +24,491 @@
 	icon_state = "speedbike_red"
 	overlay_state = "cover_red"
 
+
+// ATMOBILE AND ATMOBILE UNIQUE INTERNAL ITEMS
+
+
+
+/obj/vehicle/space/speedbike/atmos
+	name = "prototype atmos vehicle"
+	desc = "This vehicle possesses unparalleled utility for atmospheric containment and control"
+	icon_state = "atmo_bike"
+	overlay_state = "cover_atmo"
+	var/obj/machinery/portable_atmospherics/scrubber/huge/internal_scubber = null
+	var/obj/item/extinguisher/vehicle/internal_extinguisher = null
+	var/ex_out = FALSE
+	var/obj/machinery/portable_atmospherics/canister/proto/default/oxygen/CAN = null
+	var/loaded = TRUE
+	light_range = 7
+
+/obj/vehicle/space/speedbike/atmos/New()
+	. = ..()
+	internal_scubber = new /obj/machinery/portable_atmospherics/scrubber/huge(src)
+	internal_scubber.on = 1
+	internal_extinguisher = new /obj/item/extinguisher/vehicle(src)
+	internal_extinguisher.vehicle = src
+	CAN = new /obj/machinery/portable_atmospherics/canister/oxygen(src)
+//	CAN = new /obj/machinery/portable_atmospherics/canister/proto/default/oxygen(src)
+
+/obj/item/extinguisher/vehicle
+	name = "extinguisher nozzle"
+	desc = "A heavy duty nozzle attached to a massive reserve tank."
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "atmos_nozzle"
+	item_state = "nozzleatmos"
+	w_class = WEIGHT_CLASS_BULKY
+	safety = FALSE
+	max_water = 1000
+	power = 8
+	cooling_power = 10
+	recoil = 0
+	var/obj/vehicle/space/speedbike/atmos/vehicle = null
+
+/obj/item/extinguisher/vehicle/dropped(mob/user)
+	..()
+	user << "<span class='notice'>The fire hose snaps back into the [src]!</span>"
+	playsound(get_turf(src),'sound/items/change_jaws.ogg', 75, 1)
+	if(vehicle)
+		loc = vehicle
+		vehicle.ex_out = FALSE
+
+
+
+
+// ATMOBILE DATUMS
+
+
+
+/obj/vehicle/space/speedbike/atmos/buckle_mob(mob/living/M, force = 0, check_loc = 1)
+	. = ..()
+	riding_datum = new/datum/riding/space/speedbike/atmos
+	var/datum/action/innate/atmos_bike/nanoice/N = new()
+	var/datum/action/innate/atmos_bike/scrub/S = new()
+	var/datum/action/innate/atmos_bike/extinguish/E = new()
+	var/datum/action/innate/atmos_bike/flood/F = new()
+	var/datum/action/innate/atmos_bike/control/C = new()
```

Comment:
put these in the bike, create these on bike's creation, then Grant/Remove instead of constantly creating and destroying.